packages/frontpage/lib/api/post.ts

```diff
@@ -0,0 +1,84 @@
+import "server-only";
+import * as db from "../data/db/post";
+import * as atproto from "../data/atproto/post";
+import { ensureUser, getBlueskyProfile } from "../data/user";
+import { DataLayerError } from "../data/error";
+import { sendDiscordMessage } from "../discord";
+import { invariant } from "../utils";
+import { TID } from "@atproto/common-web";
+
+export type ApiCreatePostInput = {
+  title: string;
+  url: string;
+  createdAt: Date;
+};
+
+export async function createPost({
+  title,
+  url,
+  createdAt,
+}: ApiCreatePostInput) {
+  const user = await ensureUser();
+
+  const rkey = TID.next().toString();
+  try {
+    const dbCreatedPost = await db.createPost({
+      post: { title, url, createdAt },
+      rkey,
+      authorDid: user.did,
+    });
+    invariant(dbCreatedPost, "Failed to insert post in database");
+
+    const { cid } = await atproto.createPost({
+      title: title,
+      url: url,
+      rkey,
+    });
+
+    invariant(cid, "Failed to create comment, rkey/cid missing");
```

Comment:
```suggestion
    invariant(cid, "Failed to create comment, cid missing");
```

Not sure what rkey has to do with it?